### PR TITLE
Extend section about savefig.

### PR DESCRIPTION
**Pull request by t37w**
List supported formats, default, format per backend.

### CODE REVIEW

File: docs/src/output.md [hunk 1]
```diff
@@ -28,12 +28,56 @@ Plots are shown in the Atom PlotPane when possible, either when returned to the
...
 ```
 
-There are also shorthands `png(fn)`, etc.  You don't need the file extension for those.
+#### File formats supported by most graphical backends
+
+ - png
+ - svg
+ - PDF
+
+#### Default output format
```
Comment (mkborregaard) [inline docs/src/output.md]: Probably enough to add "default" after png above, keep the docs shorter.

File: docs/src/output.md [hunk 2]
```diff
@@ -28,12 +28,56 @@ Plots are shown in the Atom PlotPane when possible, either when returned to the
...
+
+ - png
+
+#### Plots supports the following output file formats
```
Comment (mkborregaard) [inline docs/src/output.md]: Or just "Supported file formats"

File: docs/src/output.md [hunk 3]
```diff
@@ -28,12 +28,56 @@ Plots are shown in the Atom PlotPane when possible, either when returned to the
...
+
+ - eps
```
Comment (mkborregaard) [inline docs/src/output.md]: Line 51 to 62 can be deleted. The information is replicated in the table.

File: docs/src/output.md [hunk 4]
```diff
@@ -28,12 +28,56 @@ Plots are shown in the Atom PlotPane when possible, either when returned to the
 
 Note that javascript-based libraries (for example: PlotlyJS) cannot be shown in the PlotPane due to issues within Atom's internals.
 
-### savefig / png
+### savefig / format
 
 Save the most recent plot, choosing file type automatically by the extension.
 
 ```julia
 savefig(filename)
```
Comment (mkborregaard) [inline docs/src/output.md]: perhaps add `savefig(plot, filename)` here?

File: docs/src/output.md [hunk 5]
```diff
@@ -28,12 +28,56 @@ Plots are shown in the Atom PlotPane when possible, either when returned to the
...
+ | png | glvisualize, gr, inspectdr, plotly, plotlyjs, pyplot |
+ | ps | gr, pyplot |
+ | svg | gr, inspectdr, pgfplots, plotly, plotlyjs, pyplot |
+ | tex | pgfplots |
+ | text | hdf5, unicodeplots |
+
+#### shorthands to save figs in a specific format
+
+There are also shorthands `png(fn)`, etc, where fn is the string containing the filename.  You don't need the file extension for those.
+Plots support 2 different versions of shorthands per format.
```
Comment (mkborregaard) [inline docs/src/output.md]: That's also true for `savefig` so I suggest adding that information as suggested in the first comment and then deleting lines 78 - 83

File: docs/src/output.md (hunk 1 above)
Comment (t37w) [reply docs/src/output.md]: Yes, this is can make the doc shorter and reduces dups.

File: docs/src/output.md (hunk 2 above)
Comment (t37w) [reply docs/src/output.md]: I'm fine, but would prefer to keep 'output' to make it clear, thus
   "Supported output file formats"

File: docs/src/output.md (hunk 3 above)
Comment (t37w) [reply docs/src/output.md]: I started with the simple list, as I wasn't sure about the readability and effort to make a table.
Then I tried a table, collecting the format per backend.
The list is easier, but the table is more useful. As you wrote in the beginning, I also assume not much changes.  Maybe more backends align to support always pdf, png, svg.

File: docs/src/output.md (hunk 4 above)
Comment (t37w) [reply docs/src/output.md]: Yes, this is missing. Thank you for the hint.

File: docs/src/output.md (hunk 5 above)
Comment (t37w) [reply docs/src/output.md]: Thank you for your feedback.  I think this section should come just after paragraph about 'savefig'.
As both commands support 1 and 2 arguments, I try to have a valid description covering both.
Followed by examples.

File: docs/src/output.md (hunk 2 above)
Comment (mkborregaard) [reply docs/src/output.md]: good point

File: docs/src/output.md (hunk 3 above)
Comment (mkborregaard) [reply docs/src/output.md]: I agree the table is nice

**Comment by mkborregaard** [comment]
Great - once you update the changes on your own fork they should become automatically updated here in the PR.

**Comment by t37w** [comment]
The reworked version is now available.

File: docs/src/output.md [hunk 6]
```diff
@@ -28,12 +28,47 @@ Plots are shown in the Atom PlotPane when possible, either when returned to the
 
 Note that javascript-based libraries (for example: PlotlyJS) cannot be shown in the PlotPane due to issues within Atom's internals.
 
-### savefig / png
+### savefig / format
 
-Save the most recent plot, choosing file type automatically by the extension.
+Plots support 2 different versions per save-command.
```
Comment (mkborregaard) [inline docs/src/output.md]: Is line 33-35 really necessary? The code and comments say exactly that.

File: docs/src/output.md [hunk 7]
```diff
@@ -28,12 +28,47 @@ Plots are shown in the Atom PlotPane when possible, either when returned to the
...
+The 1-argument command saves the most recent plot, while the 2-argument command saves the referenced plot.
+
+To save with `savefig`, choosing file type automatically by the extension of filename `fn`, use
```
Comment (mkborregaard) [inline docs/src/output.md]: Replace "`savefig` chooses file type automatically based on the file extension"

File: docs/src/output.md [hunk 8]
```diff
@@ -28,12 +28,47 @@ Plots are shown in the Atom PlotPane when possible, either when returned to the
...
+
+```julia
+savefig(fn) # save the most recent fig as fn
+savefig(plot_ref, fn) # save the fig referenced by plot_ref as fn
+```
+
+In addition, Plots offers shorthands like `png(fn)`, etc, to save into a file.
```
Comment (mkborregaard) [inline docs/src/output.md]: Delete bit after "etc."

File: docs/src/output.md [hunk 9]
```diff
@@ -28,12 +28,47 @@ Plots are shown in the Atom PlotPane when possible, either when returned to the
...
+In this case the string fn containing the filename may lack the file extension for those.
```
Comment (mkborregaard) [inline docs/src/output.md]: "may lack the file extension for those." -> "does not need a file extension".

File: docs/src/output.md [hunk 10]
```diff
@@ -28,12 +28,47 @@ Plots are shown in the Atom PlotPane when possible, either when returned to the
...
 
 ```julia
-savefig(filename)
+png(fn) # save the current fig as png with filename fn
+png(plot_ref, fn) # save the fig referenced by plot_ref as png with filename fn
 ```
 
-There are also shorthands `png(fn)`, etc.  You don't need the file extension for those.
+These examples are given using png.  The use of the others is equivalent.
```
Comment (mkborregaard) [inline docs/src/output.md]: Delete.

File: docs/src/output.md (hunk 8 above)
Comment (t37w) [reply docs/src/output.md]: OK

File: docs/src/output.md (hunk 10 above)
Comment (t37w) [reply docs/src/output.md]: OK

Comment (t37w) [reply docs/src/output.md]: Can be re-added in case a fix for my related issue report in 'Plots' is accepted.

File: docs/src/output.md (hunk 6 above)
Comment (t37w) [reply docs/src/output.md]: The sentence in line 33 I want to keep.  It is a good intro for this section.
Line 34 and 35 can be removed. I added the comments in the code example to allow copy-n-paste with a brief description.

**Comment by t37w** [comment]
The rework is available and the automatic checks are passed.

**Comment by mkborregaard** [comment]
Thanks @t37w !

**Pull request merged**